include/highfive/H5PropertyList.hpp

```diff
@@ -88,6 +99,14 @@ class PropertyList: public PropertyListBase {
     void _initializeIfNeeded();
 };
 
+const auto& getFileCreateList = [](const File& obj) {return get_plist<PropertyType::FILE_CREATE>(obj, H5Fget_create_plist);};
+const auto& getFileAccessList = [](const File& obj) {return get_plist<PropertyType::FILE_ACCESS>(obj, H5Fget_access_plist);};
+const auto& getDataSetCreateList = [](const DataSet& obj) {return get_plist<PropertyType::DATASET_CREATE>(obj, H5Dget_create_plist);};
+const auto& getDataSetAccessList = [](const DataSet& obj) {return get_plist<PropertyType::DATASET_ACCESS>(obj, H5Dget_access_plist);};
+const auto& getGroupCreateList = [](const Group& obj) {return get_plist<PropertyType::GROUP_CREATE>(obj, H5Gget_create_plist);};
+const auto& getDataTypeCreateList = [](const DataType& obj) {return get_plist<PropertyType::DATATYPE_CREATE>(obj, H5Tget_create_plist);};
+const auto& getAttributeCreateList = [](const Attribute& obj) {return get_plist<PropertyType::ATTRIBUTE_CREATE>(obj, H5Aget_create_plist);};
```

Comment:
Maybe we can reduce the number of different names:
```
FileAccessProps getAccessList(const File& obj) {
    return get_plist<PropertyType::FILE_ACCESS>(obj, H5Fget_access_plist);
}

FileCreateProps getCreateList(const File& obj) {
    return get_plist<PropertyType::FILE_CREATE>(obj, H5Fget_create_plist);
}

DatasetAccessProps getAccessList(const Dataset& obj) {
    return get_plist<PropertyType::DATASET_ACCESS>(obj, H5Dget_access_plist);
}

// ...

```

Reply:
In terms of which case to use for free functions:
https://github.com/BlueBrain/HighFive/blob/master/include/highfive/H5DataType.hpp#L252

But this shows we've been doing both (see under Functions):
https://bluebrain.github.io/HighFive/namespace_high_five.html